### ISSUE TITLE
Websocket protocol error on discord web app

### ISSUE THREAD

**Issue by jdm** [servo/servo#34520]
STR:
1. apply #34519
2. visit http://discord.com/app with `--pref dom.svg.enabled --pref dom.abort_controller.enabled`
3. with `RUST_LOG=tungstenite,net::websocket_loader`, see the following:
```
[2024-12-07T04:43:48Z TRACE net::websocket_loader] starting WS connection to wss://remote-auth-gateway.discord.gg/?v=2
[2024-12-07T04:43:48Z TRACE tungstenite::handshake::client] Request: "GET /?v=2 HTTP/1.1\r\nHost: remote-auth-gateway.discord.gg\r\nConnection: upgrade\r\nUpgrade: websocket\r\nSec-WebSocket-Version: 13\r\nSec-WebSocket-Key: Hk1LctMJrVV7vQqgE7z/pw==\r\nOrigin: https://discord.com\r\n\r\n"
[2024-12-07T04:43:48Z TRACE tungstenite::handshake::client] Client handshake initiated.
[2024-12-07T04:43:48Z TRACE tungstenite::handshake::machine] Doing handshake round.
[2024-12-07T04:43:48Z TRACE tungstenite::handshake::machine] Doing handshake round.
[2024-12-07T04:43:48Z TRACE tungstenite::handshake::machine] Doing handshake round.
[2024-12-07T04:43:48Z TRACE tungstenite::handshake::machine] Doing handshake round.
[2024-12-07T04:43:48Z WARN  net::websocket_loader] Failed to establish a WebSocket connection: Protocol(HttparseError(Version))
```

**Comment by jdm** [comment]
I'm attempting to upgrade our network stack to the latest versions of tungstenite and rustls to see if that changes anything.

**Comment by jdm** [comment]
This was not fixed by #34630.

**Comment by jdm** [comment]
From stepping through the tungstenite handshake state machine, it appears that httparse is processing a buffer of TLS-encrypted data.
